### ISSUE TITLE
Possible Memory Leak

### ISSUE THREAD

**Issue by Madscientist351** [quisquous/cactbot#2424]
![ACT memory leak](https://user-images.githubusercontent.com/78314228/106391307-a815cd00-63ba-11eb-93e8-06250caac387.png)

-->

## Description
Had previously noticed serious occasional heavy frame rate stuttering in game during Savage raids (from 141 FPS down to 60-70 fps then back to 141), decided to investigate.  Initially thought it was a DX11 issue tied to RTX GPU automatic overlocking logic.  Tried lowering fps cap via Nvidia GFCP from 141 to 131 to lower GPU load.  Played with that and no ACT, saw zero stuttering issues.  

Loaded FFXIV with same 131 fps cap, ACT w/ Cactbot ENABLED.  Noticed as went around FFXIV world today that RAM usage rose from 6 GB to approximately 12.7 GB.  As soon as I closed ACT, RAM usage dropped back down to approx 6 GB RAM.  I suspect this is the issue as potentially maxed RAM over ~3 hours raiding.

Reloaded FFXIV with same 131 fps cap, ACT, and Cactbot DISABLED.  RAM has been steady around 7 GB for last couple hours.

## Additional information
SYSTEM INFO:
WINDOWS 10 v.2004 build 19041.746
INTEL Core i5-9600K @ 5000 MHz
GIGABYTE Z390 PRO mobo
16 GB G.SKILL Ripjaws V RAM @ 3200 MHz
GIGABYTE RTX 2060 Super Gaming OC on Nvidia driver 461.40
CORSAIR TX650M PSU

If available, please include any logs or screenshots depicting the issue you're experiencing.

**Comment by quisquous** [comment]
When you say "cactbot enabled" which cactbot overlays are you running?

**Comment by Madscientist351** [comment]
Hi there,
The only overlay I was running was the raid overlay for the timer bars and
alerts w/ the TTS option.

Matt

**Comment by quisquous** [comment]
If this happens again, can you open up task manager and see which process is taking up the memory? Do you only have one overlay running? If so, if you reload that overlay via the OverlayPlugin reload button when memory is high, does that fix it?

**Comment by miaonima** [comment]
I remembered when I afked overnight and left my ACT on, the memory usage was excessive when I got back. Here is the screenshot I saved (a few weeks ago). I might have the same problem as @Madscientist351.

However, I do not know whether this is cactbot's problem, because I had raid boss trigger+timeline, eureka, jobs overlay along with a few other plugins running. Also this only happens intermittently. 
![leak](https://user-images.githubusercontent.com/35647076/107262704-b9cc2580-69f5-11eb-8ff8-cb673ac4ae57.png)

**Comment by quisquous** [comment]
Yeah, this is not that useful to just see ACT memory on its own.  Can you look at the subprocesses next time? You might also be able to reload overlays one by one and see if that frees the memory and narrow down which one it is.

**Comment by miaonima** [comment]
I had another excessive memory usage (~3GB) on CN server yesterday (raided a few hours, then afked at Limsa overnight). Reloading all overlays did not seem to free the excessive memory in my case. Disabling both OverlayPlugin.dll and CactbotOverlay.dll freed only ~300MB memory (reported by task manager), so I guess the leak is probably not from cactbot? Let me know if there is any update or suggestions.

ACT: 3.5.0.273 cactbot: 0.21.19.0 overlay: 0.16.5.0 ffxiv parsing plugin: 2.0.7.0 game: CN client patch 5.35

**Comment by valarnin** [comment]
I looked into this a bit more today while waiting for raid, and I couldn't track down any memory leak that would cause this sort of jump in memory usage.

There does seem to be an extremely minor leak (potentially) upstream in OverlayPlugin which I'll detail below, but I'm not sure it's even something that can be fixed.

Whenever an overlay handler event is triggered, it looks like CEF is JIT compiling the JS for the event that gets passed from C#, and that JIT'd code isn't ever released. This will result in approximately 3kb of memory leaked per event called. I verified this by taking snapshots of memory over an approximately hour-long period, and tracking a specific JIT'd code by ID. It persisted in memory through multiple zone changes and even a swap to a different character.

The JIT'd code remained even after manually triggering GC in dev tools. It is specific to a given overlay, so reloading the overlay will clear that memory properly (tested and verified).

Here's an example screenshot so you know what to look for if you're interested in following up on it. I'm not including the full stack because it contains my public IPv4 address but it eventually tracks back to native_context.

![image](https://user-images.githubusercontent.com/6119598/113377755-336ff800-9343-11eb-8a2d-8b6b4ed9f2c1.png)
